src/language-graphql/printer-graphql.js

```diff
@@ -122,6 +124,15 @@ function genericPrint(path, options, print) {
       return n.value;
     }
     case "StringValue": {
+      if (n.block) {
+        return concat([
+          '"""',
+          hardline,
+          n.value.replace(/"""/g, "\\$&"),
+          hardline,
```

Comment:
Doesn't this add newlines at the start and end of the block string – that is, changing the contents of the string? Is that safe?

Reply:
Actually, there's an issue with the graphql parser where both of them are the same ( https://github.com/graphql/graphql-js/issues/1188 ). I'm going to just print the value from the original source for now.

Reply:
Reading https://github.com/graphql/graphql-js/issues/1188#issuecomment-354614921 it seems like the spec says that "blank initial and trailing lines" are excluded. So I guess that’s a good thing? That means we can safely print this:

```
"""line one
line two"""
```

as

```
"""
line one
line two
"""
```

?

Not sure what "[exclude] uniform indentation with {BlockStringValue()}" means though...

Reply:
I *think* it means that code like this:

```
  """
  this text
  is indented
  """
```
is equivalent to:
```
"""
this text
is indented
"""
```

Reply:
That would be really nice! If so, we could fix indentation inside block strings as well!

Reply:
> Not sure what "[exclude] uniform indentation with {BlockStringValue()}" means though...

@lydell Example from @j-f1 is fully correct. You can read detail algorithm here:
https://github.com/facebook/graphql/pull/327/files#diff-fe406b08746616e2f5f00909488cce66R789

BTW. Would be great if you [make it compatible](https://github.com/graphql/graphql-js/blob/master/src/language/printer.js#L260-L266) with `print` function from `graphql-js`.

Reply:
@lydell @j-f1 You can also see some tests here:
https://github.com/graphql/graphql-js/blob/master/src/language/__tests__/printer-test.js#L74-L104

Reply:
Oh I didn't know graphql-js had a `print` function

Reply:
@duailibe Yes, it's part of `graphql-js` since it was first published in 2015.
It would be great if you figure out how to reuse it at least partly.

But I think most important is to have `prettier` in sync with `print` since it's de-facto standard in the GraphQL world. For example, people use `print` to print schema of remote APIs.

Reply:
The biggest differences between prettier and graphql printer is the fact that prettier is 80-columns aware and supports comments.

Thankfully, graphql is a pretty simple language so it has been easy to implement properly. I don't think that we're going to reuse any code from the project but we are definitely maintaining the same conventions that graphql is using.

The prettier graphql printer is currently used to pretty print all the graphql queries embedded inside of template literals at Facebook. So this part is rock solid. The graphql schema definition has seen less extensive use as far as I know (but correctly prints all the schema we have at Facebook, even if we don't actively format them using it).

For this specific example, we should do whatever you are doing. It seems like the only difference is to handle "inline" triple quote strings that have spaces around.

Reply:
@duailibe I forgot to mention that GraphiQL uses `print`:
![image](https://user-images.githubusercontent.com/8336157/34502761-b8d68562-f01d-11e7-8a66-55ee31f6ddca.png)
So it's would be great to ensure that by default `prettier` produces the same output.

Reply:
> The biggest differences between prettier and graphql printer is the fact that prettier is 80-columns aware and supports comments.

@vjeux Make sense 👍  and thank you for the explanation.